### PR TITLE
adding elastic box simulation

### PR DESCRIPTION
**Pull request by DiitsGp**

### CODE REVIEW

**Comment by RussTedrake** [comment]
great!   but i was hoping that we could instead add elasticity to the main lcp solver...?

File: drake/examples/Elastic_Box/Planar_CBSE_Window.URDF [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<robot
+  name="Planar_CBSE_Window">
```
Comment (tkoolen) [inline drake/examples/Elastic_Box/Planar_CBSE_Window.URDF]: What the heck is a Planar_CBSE_Window? :-)

File: drake/examples/Elastic_Box/elasticLCP.m [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    
+    z = elasticLCP(H, v, impulse, mu(contact_inds),...
+        n(contact_inds, :), D([contact_inds, contact_inds+4], :), nC, eps);
+
+    v = z(1:3);
+    q = q+h*v;
+    t = t+h;
+    
+    states = [states, [q; v]];
+    times = [times, t];
```
Comment (tkoolen) [inline drake/examples/Elastic_Box/elasticLCP.m]: can you determine the size of states and times in advance?

**Comment by tkoolen** [comment]
Looks good otherwise. Is there any documentation you can point to?

**Comment by DiitsGp** [comment]
I'll talk to John about adding this solver to the existing one, but I'm fairly certain the existing LCP is a very different formulation than the one here. Ours handles 0 restitution as well, so it could also just end up being a quick fix. Twan, the URDF is a planar brick rotated to avoid the pi/2 singularity for pitch

**Comment by RussTedrake** [comment]
i think we should not add a new urdf for that.  at the least you can pass in an rpy to addRobotFromURDF which would rotate the model.  but better would be to use the quaternion floating base.

**Comment by tkoolen** [comment]
Agree with Russ. By the way, I was referring to the fact that nobody will understand that a Planar_CBSE_Window is in fact a planar brick rotated to avoid the pi/2 singularity for pitch (I still don't see the connection). Naming things in an understandable way matters!

**Comment by DiitsGp** [comment]
Is it ok to have two side-by-side LCP solves in TSRBM as a temporary fix? Integrating Anitescu/Potra with Stewart/Trinkle isn't very straightforward, and I'm not certain how long it will take to have one LCP that can handle everything.

**Comment by RussTedrake** [comment]
yes, definitely. in that case, I would make it two separate functions 
(with logic that decides to call your function instead of 
solveLCP/solveLCPmex near the top).

DiitsGp wrote:

**Pull request closed without merging**